api/envoy/extensions/filters/network/http_connection_manager/v3/http_connection_manager.proto

```diff
@@ -816,7 +816,7 @@ message ScopedRds {
       [(validate.rules).message = {required: true}];
 }
 
-// [#next-free-field: 6]
+// [#next-free-field: 7]
 message HttpFilter {
```

Comment:
Not specific to gRPC: Should we also add this "optional" functionality to [upstream filters](https://github.com/envoyproxy/envoy/blob/1942151aafceee50630572c82cc40d220ec4a63c/api/envoy/config/cluster/v3/filter.proto#L18) in cluster? Then if one day Envoy decided to implement this feature, they will have all proto changes needed.

Reply:
In general, I think it would be useful to do this for all types of filters (also including L4 filters and listener filters).  But right now, we are primarily concerned about HTTP filters, so let's focus this PR on that functionality.